src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -499,24 +540,33 @@ final case class AgreementApiServiceImpl(
             s"Descriptor ${agreement.descriptorId} Service ${agreement.eserviceId} not found for agreement ${agreement.id}"
           )
           Future.unit
-        }(updateAgreement(agreement))
+        }(updateAgreement(agreement.toManagement))
 
-    val updatableStates = allowedStateTransitions.map { case (startingState, _) => startingState }.toList
+    val updatableStates = allowedStateTransitions.map { case (startingState, _) => startingState }.toSeq
 
-    def eServiceContainsAttribute(attributeId: UUID)(eService: CatalogManagement.EService): Boolean = {
+    def eServiceContainsAttribute(attributeId: UUID)(eService: CatalogItem): Boolean = {
       val certified = eService.descriptors.flatMap(_.attributes.certified)
       val declared  = eService.descriptors.flatMap(_.attributes.declared)
       val verified  = eService.descriptors.flatMap(_.attributes.verified)
       (certified ++ declared ++ verified)
-        .flatMap(attr => attr.single.map(_.id).toSeq ++ attr.group.traverse(_.map(_.id)).flatten)
+        .flatMap(attrs =>
+          attrs match {
+            case single: SingleAttribute => Seq(single.id.id)
+            case group: GroupAttribute   => group.ids.map(_.id)
+          }
+        )
```

Comment:
Very very minor code improvement
```suggestion
        .flatMap {
          case single: SingleAttribute => Seq(single.id.id)
          case group: GroupAttribute   => group.ids.map(_.id)
        }
```